MessageCenter/Classes/Views/Chatting/GroupChannelChattingViewController.swift

```diff
@@ -631,6 +629,33 @@ class GroupChannelChattingViewController: UIViewController, SBDConnectionDelegat
         self.present(mediaUI, animated: true, completion: nil)
     }
     
+    @objc private func sendFileMessage() {
+        let status = PHPhotoLibrary.authorizationStatus()
+        if status == .authorized {
+            openPicker()
+        } else {
+            PHPhotoLibrary.requestAuthorization { status in
+                switch status {
+                case .authorized:
+                    self.openPicker()
```

Comment:
This could fail since `requestAuthorization` might call the handler on an arbitrary serial queue.
```suggestion
DispatchQueue.main.async {
                    self.openPicker()
}
```